backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,278 @@
+            ID_DEFAULT, PERSON_B, TIME_1400ET, TIME_1800ET, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC =
+        new Participant(
+            ID_DEFAULT, PERSON_C, TIME_1400ET, TIME_2000ET, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+
+    Match match = query.findMatch(Arrays.asList(participantA, participantB), participantC);
+
+    assertThat(match.getFirstParticipant().getUsername()).isEqualTo(PERSON_C);
+    assertThat(match.getSecondParticipant().getUsername()).isEqualTo(PERSON_A);
+    assertThat(match.getDuration()).isEqualTo(DURATION_30_MINUTES);
+  }
+
+  @Test
+  public void barelyNotCompatible() {
+    // Two participants barely NOT compatible availability (edge case, need >15 minutes padding)
+    Participant participantA =
+        new Participant(
+            ID_DEFAULT, PERSON_A, TIME_1400ET, TIME_1500ET, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB =
+        new Participant(
+            ID_DEFAULT, PERSON_B, TIME_1400ET, TIME_1800ET, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+
+    Match match = query.findMatch(Arrays.asList(participantA), participantB);
+
+    assertThat(match).isNull();
+  }
+
+  @Test
+  public void barelyCompatible() {
+    // Two participants barely compatible availability (edge case, need >15 minutes padding for
+    // compatibility)
+    Participant participantA =
+        new Participant(
+            ID_DEFAULT, PERSON_A, TIME_1400ET, TIME_1501ET, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB =
+        new Participant(
+            ID_DEFAULT, PERSON_B, TIME_1400ET, TIME_1800ET, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+
+    Match match = query.findMatch(Arrays.asList(participantA), participantB);
+
+    assertThat(match.getFirstParticipant().getUsername()).isEqualTo(PERSON_B);
+    assertThat(match.getSecondParticipant().getUsername()).isEqualTo(PERSON_A);
+    assertThat(match.getDuration()).isEqualTo(DURATION_45_MINUTES);
+  }
+
+  @Test
+  public void compatibleDiffTimezones() {
+    // Two participants, A in ET timezone and B in PT timezone and compatible
+    Participant participantA =
+        new Participant(
+            ID_DEFAULT, PERSON_A, TIME_1400ET, TIME_1600ET, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB =
+        new Participant(
+            ID_DEFAULT, PERSON_B, TIME_1100PT, TIME_1600PT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+
+    Match match = query.findMatch(Arrays.asList(participantA), participantB);
+
+    assertThat(match.getFirstParticipant().getUsername()).isEqualTo(PERSON_B);
+    assertThat(match.getSecondParticipant().getUsername()).isEqualTo(PERSON_A);
+    assertThat(match.getDuration()).isEqualTo(DURATION_30_MINUTES);
+  }
+
+  @Test
+  public void compatibleWithoutConsideringTimezone() {
+    // Two participants, A in ET timezone and B in PT timezone, compatible if not considering
+    // timezones, compatible if considering their different timezones
```

Comment:
They are compatible *only* because their timezones have been considered. Otherwise 1400 - 1500 and 1100 - 1200 wouldn't overlap.